Registration and forced import

***In GitLab by @woutdenolf on Nov 5, 2021, 14:01 GMT+1:***

```python
ERROR    Orange.canvas.registry.discovery:discovery.py:259 Error while importing 'orangecontrib.darfix.widgets.zsum'. The widget will not be shown.
Traceback (most recent call last):
  File "/home/denolf/dev/orange3_henri/Orange/canvas/registry/discovery.py", line 257, in iter_widget_descriptions
    module = asmodule(name)
  File "/home/denolf/dev/orange3_henri/Orange/canvas/registry/discovery.py", line 484, in asmodule
    return __import__(module, fromlist=[""])
  File "/home/denolf/dev/darfix/orangecontrib/darfix/widgets/zsum.py", line 35, in <module>
    from darfix.core.process import ZSum
  File "/home/denolf/dev/darfix/darfix/core/process.py", line 49, in <module>
    class IgnoreProcess(Task, input_names=["dataset"], output_names=["dataset"]):
  File "/home/denolf/dev/ewokscore/ewokscore/task.py", line 90, in __init_subclass__
    super().__init_subclass__(**kwargs)
  File "/home/denolf/dev/ewokscore/ewokscore/registration.py", line 27, in __init_subclass__
    raise NotImplementedError(
NotImplementedError: Registry name darfix.core.process.IgnoreProcess is already taken by <class 'darfix.core.process.IgnoreProcess'>
```

**Assignees:** @woutdenolf

*Migrated from GitLab: https://gitlab.esrf.fr/workflow/ewoks/ewokscore/-/issues/15*

--- a/ewokscore/registration.py
+++ b/ewokscore/registration.py
@@ -21,6 +21,8 @@ class Registered:
             registry_name = utils.qualname(cls)
         ecls = cls._SUBCLASS_REGISTRY.get(registry_name)
         if ecls is not None:
+            if utils.qualname(cls) == utils.qualname(ecls):
+                return
             raise NotImplementedError(
                 f"Registry name {registry_name} is already taken by {repr(ecls)}"
             )

